homeassistant/components/music_assistant/services.py

```diff
@@ -0,0 +1,91 @@
+"""Custom services for the Music Assistant integration."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, cast
+
+from music_assistant_client.helpers import searchresults_as_compact_dict
+from music_assistant_models.enums import MediaType
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import (
+    HomeAssistant,
+    ServiceCall,
+    ServiceResponse,
+    SupportsResponse,
+    callback,
+)
+from homeassistant.exceptions import HomeAssistantError
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN
+
+if TYPE_CHECKING:
+    from music_assistant_client import MusicAssistantClient
+
+    from . import MusicAssistantConfigEntry
+
+SERVICE_SEARCH = "search"
+ATTR_MEDIA_TYPE = "media_type"
+ATTR_SEARCH_NAME = "name"
+ATTR_SEARCH_ARTIST = "artist"
+ATTR_SEARCH_ALBUM = "album"
+ATTR_LIMIT = "limit"
+ATTR_LIBRARY_ONLY = "library_only"
+
+
+@callback
+def get_music_assistant_client(hass: HomeAssistant) -> MusicAssistantClient:
+    """Get the (first) Music Assistant client from the (loaded) config entries."""
+    entry: MusicAssistantConfigEntry
+    for entry in hass.config_entries.async_entries(DOMAIN, False, False):
+        if entry.state != ConfigEntryState.LOADED:
+            continue
+        return entry.runtime_data.mass
+    raise HomeAssistantError("Music Assistant is not loaded")
```

Comment:
In theory, users can now setup two music_assistant integrations. Why don't we ask the config entry as parameter in this function call? Then we don't have to loop over all the config entries and find a loaded one

Reply:
I think we need to have both actually - I can imagine us landing in a situation where we want to have the mass client but we do not have access to the config entry. Media browser is a good example.

So this helper function should optionally take a config entry (id) and if that is omitted, we do the loop to take the first loaded one.

Reply:
> I can imagine us landing in a situation where we want to have the mass client but we do not have access to the config entry.

But we need to know what mass client we need to pick right? You can connect to two MASS with 2 different libraries right?

Reply:
Problem is... how?
The services are registered from async_setup so we dont have access to the config entry there.
All we can do is loop through the entries and pick one (or maybe even try all if there are multiple).

Or do you mean actually requiring the user to provide the config entry id ?
That is not super user friendly.

BTW: the same issue exists in Hue and some other integrations that provide domain services.

Reply:
nvm - learned about the integration selector so all good now.